src/java.base/windows/native/libjava/java_props_md.c

```diff
@@ -128,15 +128,19 @@ getEncodingInternal(LCID lcid)
     return ret;
 }
 
-static char* getConsoleEncoding()
+static char* getConsoleEncoding(BOOL output)
 {
     size_t buflen = 16;
     char* buf = malloc(buflen);
     int cp;
     if (buf == NULL) {
         return NULL;
     }
-    cp = GetConsoleCP();
+    if (output) {
+        cp = GetConsoleOutputCP();
+    } else {
+        cp = GetConsoleCP();
```

Comment:
While we are here, shouldn't this handle the error case where GetConsoleXXX returns 0 for error. We've seen bugs where "cp0" is used and I think this suggests that it must fail in some environments.

Reply:
Ah, you mentioned it before which I forgot! Handled the error case too.

Reply:
How is NULL handled in the callers?

Reply:
Those system properties will be substituted with `native.encoding`
https://github.com/openjdk/jdk/blob/dc5c4148c70ca43d0a69c326e14898adca2f0bae/src/java.base/share/classes/jdk/internal/util/SystemProps.java#L98